### PR TITLE
kjv: init at unstable-2018-12-25

### PR DESCRIPTION
**Pull request by jtobin**
###### Motivation for this change

Simply adds bonibon's `kjv` to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/kjv/default.nix
Comment (jonringer) [inline pkgs/applications/misc/kjv/default.nix]: You should be able to use the Makefile they provide, just pass it the right makeFlags
```suggestion
  makeFlags = [
    "PREFIX=$(out)"
  ];
```
should be able to remove your custom builder.sh :)

File: pkgs/top-level/all-packages.nix
Comment (jonringer) [inline pkgs/top-level/all-packages.nix]: nit :)
```suggestion
  kjv = callPackage ../applications/misc/kjv { };
```

File: pkgs/applications/misc/kjv/default.nix
Comment (jonringer) [inline pkgs/applications/misc/kjv/default.nix]: Usually you want to keep with versions that they release, if they haven't (like in this case), then just give it the date of the last update
```suggestion
  version = "20190311";
```

**Review by jonringer — COMMENTED** [review]
A few suggestions, but i was able to build the package.

```
./result/bin/kjv --help
usage: kjv [flags] [reference...]

  -l      list books
  -W      no line wrap
  -h      show help
...
```

**Comment by jtobin** [comment]
@jonringer Thanks for the careful review!  I've made all the changes you suggested.

**Review by jonringer — COMMENTED** [review]
just a little more cleanup, doesn't change the closure though :)

```
$ nix-store -q --tree ./result # after
/nix/store/wzdw64avynfad1mlr11sb7qa02v86jdz-kjv
+---/nix/store/7bkv33x88dc64yhjvfvjvh0fqyhlliwj-bash-4.4-p23
    +---/nix/store/bjmg1g133m9xwxa0iy5inp2snvnfg151-glibc-2.27
    |   +---/nix/store/bjmg1g133m9xwxa0iy5inp2snvnfg151-glibc-2.27 [...]
    +---/nix/store/7bkv33x88dc64yhjvfvjvh0fqyhlliwj-bash-4.4-p23 [...]
```
```
$ nix-store -q --tree ./result # before
/nix/store/av5h5mj4w1cxc0kxm31ndawn4pcrc18g-kjv
+---/nix/store/7bkv33x88dc64yhjvfvjvh0fqyhlliwj-bash-4.4-p23
    +---/nix/store/bjmg1g133m9xwxa0iy5inp2snvnfg151-glibc-2.27
    |   +---/nix/store/bjmg1g133m9xwxa0iy5inp2snvnfg151-glibc-2.27 [...]
    +---/nix/store/7bkv33x88dc64yhjvfvjvh0fqyhlliwj-bash-4.4-p23 [...]
```

**Comment by jonringer** [comment]
You'll have to sit tight until someone with commit access reviews :)

File: pkgs/applications/misc/kjv/default.nix
Comment (jb55) [inline pkgs/applications/misc/kjv/default.nix]: `repo = name` works just fine here

```suggestion
    repo = name;
```

**Review by jb55 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/kjv/default.nix
Comment (jb55) [inline pkgs/applications/misc/kjv/default.nix]: try this instead

```suggestion
    inherit (src.meta) homepage;
```

**Review by jb55 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/kjv/default.nix
Comment (jonringer) [reply pkgs/applications/misc/kjv/default.nix]: although better on the DRY principle, there's a lot of cases when im fixing package that i just want to be able to copy and paste where the thing is located.

Comment (jb55) [reply pkgs/applications/misc/kjv/default.nix]: check out this:

```bash
nix-homepage () { nix-instantiate --eval --expr "with import <nixpkgs> {}; pkgs.$1.meta.homepage" | sed 's,^",,;s,"$,,' }
```

Comment (jonringer) [reply pkgs/applications/misc/kjv/default.nix]: still seems quite removed from the convenience and explicitness of a url located with the other meta information.

I'm not saying one is strictly better, I have a personal leaning toward to a raw url or string.

Comment (jb55) [reply pkgs/applications/misc/kjv/default.nix]: yeah I'm cool with either, I don't have a strong opinion on it.

**Comment by jtobin** [comment]
@jb55 Thanks for the review!  I kept the raw URL for the homepage, but simplified the repo attribute per your suggestion.

**Comment by jtobin** [comment]
(Pinging anyone with push access to take another look at this.)

**Comment by jonringer** [comment]
@marsam do you mind take a look at this? :). Should be ready to go

File: pkgs/applications/misc/kjv/default.nix
Comment (marsam) [inline pkgs/applications/misc/kjv/default.nix]:
```suggestion
  pname = "kjv";
```

File: pkgs/applications/misc/kjv/default.nix
Comment (marsam) [inline pkgs/applications/misc/kjv/default.nix]: please use `unstable-YYYY-MM-DD`. See https://nixos.org/nixpkgs/manual/#sec-package-naming

File: pkgs/applications/misc/kjv/default.nix
Comment (marsam) [inline pkgs/applications/misc/kjv/default.nix]: why package a fork instead of the main repository? https://github.com/bontibon/kjv

File: pkgs/applications/misc/kjv/default.nix
Comment (marsam) [inline pkgs/applications/misc/kjv/default.nix]: please don't use `name` here

File: pkgs/applications/misc/kjv/default.nix
Comment (jonringer) [reply pkgs/applications/misc/kjv/default.nix]: there should be a link to that section on the contributing doc....

File: pkgs/applications/misc/kjv/default.nix
Comment (jonringer) [reply pkgs/applications/misc/kjv/default.nix]: it contains another book in the bible

Comment (jonringer) [reply pkgs/applications/misc/kjv/default.nix]: why they didn't do a PR to merge it back in, idk

Comment (marsam) [reply pkgs/applications/misc/kjv/default.nix]: I think the end-user could [override it](https://nixos.org/nixos/nix-pills/nixpkgs-overriding-packages.html) with the [patch](https://github.com/LukeSmithxyz/kjv/commit/b92b7622285d10464f9274f11e740bef90705bbc)
```nix
  kjv = pkgs.kjv.overrideAttrs (attrs: {
    patches = attrs.patches ++ [ /path/to/apocrypha.patch ];
  });
```

Comment (jtobin) [reply pkgs/applications/misc/kjv/default.nix]: It seems fine and simpler to me to package the fork, and would prefer to stick with that.  Just let me know if you don't consider that to be acceptable.

**Comment by jtobin** [comment]
@marsam Thanks for the review!  I've made most of your suggestions, and left a comment re: packaging the fork.

File: pkgs/applications/misc/kjv/default.nix
Comment (worldofpeace) [reply pkgs/applications/misc/kjv/default.nix]: How about just fetching the patches from the fork and applying them?

File: pkgs/applications/misc/kjv/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/kjv/default.nix]:
```suggestion
    repo = pname;
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]
See comments.

Also can you change the commit msg `Add maintainer info for Jared Tobin.` to
```
maintainers: add Jared Tobin
```

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build kjv

**Comment by jtobin** [comment]
@worldofpeace Thanks for the review -- I've made the changes you (and, re: avoiding the fork, also @marsam) requested.

Just let me know if anything else seems worth patching up.  In particular, I added a manual `installPhase` definition to account for the lack of a phony `install` target in the canonical repo's Makefile, and I wouldn't be surprised if there were some more idiomatic way of handling that.

File: pkgs/applications/misc/kjv/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/kjv/default.nix]: We mean't to fetch this commit also https://github.com/LukeSmithxyz/kjv/commit/f4ad73539eb73f1890f4b791d8d38dd95900a4a4

Then you won't need this.

File: pkgs/applications/misc/kjv/default.nix
Comment (worldofpeace) [inline pkgs/applications/misc/kjv/default.nix]: You can inline this in `patches` like
```
patches = [
  (fetchpatch {
    url = ...
    sha256 = ...
  })
];
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/kjv/default.nix
Comment (jtobin) [reply pkgs/applications/misc/kjv/default.nix]: Fetching the Makefile-tweaking patch doesn't actually strike me as being desirable here, though.  The patch that I've fetched is the only interesting addition that the fork makes; using my own `installPhase` here actually seems a bit cleaner.

File: pkgs/applications/misc/kjv/default.nix
Comment (jtobin) [reply pkgs/applications/misc/kjv/default.nix]: I tend to favour explicit bindings (in the spirit, if not the letter, of [A-normal form](https://en.wikipedia.org/wiki/A-normal_form)).

**Comment by jtobin** [comment]
@worldofpeace Thanks for following up!  I left some comments, but will happily defer to you here if my approach isn't acceptable.

**Comment by worldofpeace** [comment]
I'd have to disagree with https://github.com/NixOS/nixpkgs/pull/64565#discussion_r306473672.

The commit adds an install target https://github.com/LukeSmithxyz/kjv/commit/f4ad73539eb73f1890f4b791d8d38dd95900a4a4#diff-b67911656ef5d18c4ae36cb6741b7965R16
which in nixpkgs would always be preferred instead of doing something manually.

**Comment by jtobin** [comment]
@worldofpeace All good -- I've added the install target patch as well.

**Review by worldofpeace — APPROVED** [review]
LGTM :+1: 

I'd also like to mention it's best avoided to start your PR branch directly on `master`.
See https://nixos.org/nixpkgs/manual/#submitting-changes-making-patches

**Pull request merged**
